client/src/test/java/org/apache/cassandra/sidecar/client/retry/BasicRetryPolicyTest.java

```diff
@@ -0,0 +1,376 @@
+    }
+
+    @Test
+    void testRetriesWithNotAcceptableStatusCode()
+    {
+        when(mockResponse.statusCode()).thenReturn(NOT_ACCEPTABLE.code());
+        testWithRetries(mockRequest, mockResponse, null, 8, 50, true);
+    }
+
+    @ParameterizedTest(name = "{index} => canRetryOnADifferentHost={0}")
+    @ValueSource(booleans = { true, false })
+    void testRetriesWithServiceUnavailableStatusCode(boolean canRetryOnADifferentHost)
+    {
+        when(mockResponse.statusCode()).thenReturn(SERVICE_UNAVAILABLE.code());
+        testWithRetries(mockRequest, mockResponse, null, 5, 300, canRetryOnADifferentHost);
+    }
+
+    @Test
+    void testRetriesWithServiceUnavailableStatusCodeWithRetryAfterHeader()
+    {
+        when(mockResponse.statusCode()).thenReturn(SERVICE_UNAVAILABLE.code());
+        headersMap.put("Retry-After", Collections.singletonList("5")); // 5 seconds -> 5,000 millis
+        testWithRetries(mockRequest, mockResponse, null, 5, 1000, 5000, false);
+        testWithRetries(mockRequest, mockResponse, null, 5, 1000, 0, true);
+    }
+
+    @Test
+    void testRetriesWithServiceUnavailableStatusCodeWithInvalidRetryAfterHeader()
+    {
+        when(mockResponse.statusCode()).thenReturn(SERVICE_UNAVAILABLE.code());
+        headersMap.put("Retry-After", Collections.singletonList("one-thousand-seconds"));
+        testWithRetries(mockRequest, mockResponse, null, 5, 200, 200, false);
+        testWithRetries(mockRequest, mockResponse, null, 5, 200, 0, true);
+    }
+
+    @Test
+    void testRetriesWithAcceptedStatusCode() throws ExecutionException, InterruptedException
+    {
+        when(mockResponse.statusCode()).thenReturn(ACCEPTED.code());
+        RetryPolicy retryPolicy = new BasicRetryPolicy(1);
+        CompletableFuture<HttpResponse> future = new CompletableFuture<>();
+        AtomicInteger retryActionCalls = new AtomicInteger(0);
+        int maxRetries = 50;
+        for (int currentAttempt = 1; currentAttempt <= maxRetries; currentAttempt++)
+        {
+            retryPolicy.onResponse(future, mockRequest, mockResponse, null, currentAttempt, true,
+                                   (attempts, retryDelayMillis) -> {
+                                       retryActionCalls.incrementAndGet();
+                                       assertThat(attempts).isEqualTo(1);
+                                       assertThat(retryDelayMillis).isEqualTo(0);
+                                   });
+        }
+
+        when(mockResponse.statusCode()).thenReturn(OK.code());
+
+        retryPolicy.onResponse(future, mockRequest, mockResponse, null, 1, false,
+                               (attempts, retryDelayMillis) -> fail("Should not retry"));
+        future.join();
+        assertThat(future.isDone()).isTrue();
+        assertThat(future.get()).isSameAs(mockResponse);
+        assertThat(retryActionCalls.get()).isEqualTo(maxRetries);
+    }
+
+    private static Stream<Arguments> clientStatusCodeArguments()
```

Comment:
nit: All `4XX` are client error. The stream returned is not the complete set. The original implementation only returns  the established client error codes, which is a subset. That being said, it is rarely a good idea to have custom status code. I only suggest to cover all 4XX code from the perspective of testing, because all of them technically represent client errors, regardless of the predefined or the custom. 

```suggestion
    private static Stream<Arguments> clientStatusCodeArguments()
    {
        return IntStream.range(400, 500).boxed().map(Arguments::of);
    }
```